src/core/router/transports/ssu/packet.cc

```diff
@@ -893,7 +903,9 @@ std::unique_ptr<SSUPeerTestPacket> SSUPacketParser::ParsePeerTest() {
   auto packet = std::make_unique<SSUPeerTestPacket>();
   packet->SetNonce(Read<std::uint32_t>());
   std::uint8_t const size(Read<std::uint8_t>());
-  packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
+  packet->SetIPAddressSize(size);
+  if (size)
+    packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));
```

Comment:
>packet->SetIPAddressSize(size);
>if (size)
>  packet->SetIPAddress(core::BytesToAddress(ReadBytes(size), size));

This can set an invalid size and assumes that if the value is not null, then it's valid - which is not true. This is not a huge issue because an invalid size will throw but, if the bytes to address convertor doesn't throw, we can save time by calling the size setter *after* we have valid data.

Note: I thought these setters were removed in place of a real address object but I apparently had only removed them in my mind, as noted in https://github.com/monero-project/kovri/issues/140#issuecomment-369775064 :laughing:

Reply:
> This can set an invalid size and assumes that if the value is not null, then it's valid - which is not true. This is not a huge issue because an invalid size will throw but, if the bytes to address convertor doesn't throw, we can save time by calling the size setter after we have valid data.

You're right, this was my mistake, I was relying on the the throw behavior for invalid sizes. I'll do the correct checks before setting IP address size.

> I thought these setters were removed in place of a real address object but I apparently had only removed them in my mind

Saw that you are using `boost::asio::ip::address`, but wasn't sure how to use its member functions in this context without the extra getter + setter.